src/monmove.cpp

```diff
@@ -271,6 +277,109 @@ float monster::rate_target( Creature &c, float best, bool smart ) const
     return INT_MAX;
 }
 
+std::map<item*, const tripoint> monster::find_loot_in_radius(const tripoint& target, int radius) {
+    bool steals_food = has_flag(MF_STEALS_FOOD);
+    bool steals_shiny = has_flag(MF_STEALS_SHINY);
+    bool suitable = false;
+    std::map<item*, const tripoint> lootable;
+
+    //Check all tiles nearby within radius
+    for (const tripoint &p : g->m.points_in_radius(target, radius)) {
+        //Check if we can see some items on the tile
+        if (g->m.sees_some_items(p, *this)) {
+            //A candidate tile with some items!
+            auto items = g->m.i_at(p);
+            //Check all items on tile
+            for (item &itm : items) {
+                suitable = false;
+
+                //Does monster try to steal food and is it actually food?
+                if (steals_food && (itm.is_comestible() && itm.get_comestible()->comesttype == "FOOD")) {
+                    if (!itm.rotten())
+                        suitable = true;
+
+                    //TODO: Sanity checks here on what we're trying to loot.
+                }
+
+                //Does monster steal shiny items? 
+                if (steals_shiny &&
+                    (itm.made_of(material_id("gold")) ||
+                    itm.made_of(material_id("silver")) ||
+                    itm.made_of(material_id("diamond")) || 
+                    itm.made_of(material_id("platinum")))) {
+
+                    //TODO: Sanity checks on what we're trying to loot here
+
+                    suitable = true;
+                }
+           
+
+                if (suitable)
+                    lootable.insert(std::make_pair(&itm, p));
+            }
+        }
+    }
+
+    return lootable;
+}
+
+bool monster::eat_from_inventory() {
+    auto it = inv.begin();
+    for (auto& itm : inv) {
```

Comment:
please remove this auto in favor of the actual class name; in this case it appears to be item